### PR TITLE
946 add redirect to DMP

### PR DESCRIPTION
**Pull request by jbperidypathtech**
This fixes issue #946.

How to test the feature:

- Get this picture
image

Screenshot/video of feature:
image
image

This PR also fixes a few edge cases for incident reports with no DisDate and only "start"
image

### CODE REVIEW

File: frontend/src/components/MapView/utils.ts [hunk 1]
```diff
@@ -87,29 +88,40 @@ export const downloadToFile = (
...
+              featureInfoProps[item].labelMap,
+            ),
+            coordinates,
+          },
+        };
+      },
+      {
+        dmpDisTyp: properties.DisTyp,
+        dmpSubmissionId: properties.submission_id,
+      },
```
Comment (ericboucher) [inline frontend/src/components/MapView/utils.ts:123]: This piece of code feels a bit too specific. Maybe wrap it around a util to handle metadata?
Remember that DMP and Kobo stuff are only used for Cambodia atm

Comment (jbperidypathtech) [reply frontend/src/components/MapView/utils.ts]: Indeed I just change it

File: frontend/src/config/cambodia/layers.json [hunk 2]
```diff
@@ -1713,6 +1713,14 @@
       "status": {
         "type": "text",
         "dataTitle": "Form status"
+      },
+      "submission_id": {
```
Comment (ericboucher) [inline frontend/src/config/cambodia/layers.json:1717]: Maybe metadata should just keep their initial name and be a bit more lenient in the netadata type? Wdyt?
No strong feelings, open to suggestions.

cc @laurentS interested in your input as well

**Comment by wadhwamatic** [comment]
This works well for flood and drought reports. After Eric's fix, the incident reports load without an error, but the hyperlink in the tooltip doesn't appear.

Screenshot 2023-11-03 at 11 51 22

File: frontend/src/config/cambodia/layers.json (hunk 2 above)
Comment (jbperidypathtech) [reply frontend/src/config/cambodia/layers.json:1717]: I transform submission_id in dmpSubmissionId to use camel case in the typescript code, and add also dmp prefix to be more specific about what is submission id. But if you rather like to use submission_id directly it is ok for me.

"be a bit more lenient in the metadata type" do you suggest to add a type to control and raise an error if it is a number rather than a string for instance? It is possible but it appears a little overkill for me. If required I can add this control.

**Comment by jbperidypathtech** [comment]
For incident reports it was an edge case when DisTyp is both data (to display in popup) and metadata (to not display in popup) I changed a little bit the logic to cover this case.

File: frontend/src/config/cambodia/layers.json (hunk 2 above)
Comment (ericboucher) [reply frontend/src/config/cambodia/layers.json:1717]: No I actually meant "do not create a type for metadata and return all additional info that's not "data". This way people can simply decide to use the extra data in their logic, or not. :)
I am ok with the current implementation as well

Comment (ericboucher) [reply frontend/src/config/cambodia/layers.json:1717]: So thinking about it a bit more we could actually revamp this a bit to also take into account the "do_not_display" behavior by making the dataTitle optional.

Then:
Option 1 - keys in `feature_info_props` would become data and all additional info would become metadata
Option 2 - keys in `feature_info_props` accept a "metadata" flag to split info between data and metadata like you're doing now.

Wdyt?

Comment (jbperidypathtech) [reply frontend/src/config/cambodia/layers.json:1717]: I think option 2 is better to covert the case when we want both to display the data in the popup and use the data in the metadata while remaining generic. But I bring a small simplification in the layer.json to simplify metadata object.

**Review by ericboucher — APPROVED** [review]

File: frontend/src/config/types.ts [hunk 3]
```diff
@@ -372,6 +372,7 @@ interface FeatureInfoProps {
   type: DataType;
   dataTitle: string;
   labelMap?: { [key: string]: string };
+  metadata?: string;
```
Comment (ericboucher) [inline frontend/src/config/types.ts:375]: could this be types with the keys of `PopupMetaData`? This will make it easier to understand what to change in the future
